cmake/compilers.cmake

```diff
@@ -38,8 +38,9 @@ function(CHECK_COMPILERS)
     endif()
 endfunction()
 
-function(ADJUST_COMPILER_WARNINGS)
+function(ADJUST_METAL_COMPILER_WARNINGS)
```

Comment:
had to rename because this conflicted with UMD's function that has the same name.
UMD version has to be removed, but out of scope for this PR

Reply:
It is removed in latest UMD, but metal might not point to it yet.